### DISCUSSION TITLE
ESP-NOW add nodes to a network

### DISCUSSION THREAD

**Discussion by orbitcoms** [Mair/esp32-course#94] [Course Qeustions] [answered]
Hi Mair,

This is related to the ESP-NOW videos regarding adding new nodes to a network.

I have an application that has up to 3 nodes on a vehicle. One node is the main controller and the other 2 are remote sensors

I was thinking to pair them that I could have the push button that places them in "Pair" mode where the controller sends a broadcast "Controller Ping" and the remote units send broadcast "Sensor Ping". After they pair, the user reboots the devices and they run paired together. Alternately I could use the button to run a wifi web server and user manually enters the MAC addresses.

The question I have is related to my last post.
Can I have unlimited number of vehicles all with say 3 nodes in their network running?

I am not sure if the 20 device limit is a resource issue on the ESP32 (storing the MAC addresses etc), in which case I suspect it will work ok, or whether it is a wifi radio limitation?

Thanks

**Comment by Mair** [comment, marked as answer]
It should be fine

**Reply by orbitcoms** [reply]
Thanks Mair

**Comment by gmag11** [comment]
Hi, I am developing an Arduino ESP-NOW library to use on my projects. It is not finished yet but it is (should be) working in ESP32. I still have to test in on ESP8266. It has unlimited peers, indeed you do not have to worry about registering them. All is done internally and automatically.

https://github.com/gmag11/QuickESPNow

There is no documentation yet, but examples are self explanatory.
